[Linux][SDL2] Building process is broken in render.cpp

**Describe the bug**
After AVX2 was added to main code, I cannot get Dosbox-X to be build. Every single time, build process stop while processing render.cpp.

Error log start with:

```
render.cpp:172:67: warning: AVX vector return without AVX enabled changes the ABI [-Wpsabi]
  172 |                 __m256i v = _mm256_loadu_si256((const __m256i*)src);
      |                                                                   ^
In file included from /usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/include/immintrin.h:53,
                 from /usr/include/SDL2/SDL_cpuinfo.h:86,
                 from /usr/include/SDL2/SDL.h:38,
                 from ../../include/sdlmain.h:10,
                 from render.cpp:35:
/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/include/avx2intrin.h:433:1: error: inlining failed in call to 'always_inline' 'int _mm256_movemask_epi8(__m256i)': target specific option mismatch
```

**To Reproduce**
Steps to reproduce the behavior:
1. Grab latest git code
2. Use this configure line: ./configure --enable-core-inline --disable-debug --enable-avcodec --prefix=/usr --enable-sdl2
3. Run make and wait.

**Expected behavior**
Build process going on.

**Environment (please complete the following information):**
 - Archlinux
 - commit SHA1: b5310f1c3671bfe1b0226fd75a8703053fa4ecdb

Attaching crash log.
[dosbox-x-crash.log](https://github.com/joncampbell123/dosbox-x/files/5286470/dosbox-x-crash.log)

--- a/src/dosbox.cpp
+++ b/src/dosbox.cpp
@@ -100,7 +100,7 @@
 #include <list>
 
 /*===================================TODO: Move to it's own file==============================*/
-#if defined(__SSE__) && !(defined(_M_AMD64) || defined(__e2k__))
+#ifdef __SSE__
 bool sse2_available = false;
 bool avx2_available = false;
 
@@ -114,7 +114,7 @@ bool avx2_available = false;
     __asm mov d, edx
 # endif /* _MSC_VER */
 
-void CheckSSESupport()
+static void CheckX86ExtensionsSupport()
 {
 #if defined(__GNUC__) && !defined(EMSCRIPTEN)
     sse2_available = __builtin_cpu_supports("sse2");
@@ -1357,8 +1357,8 @@ void DOSBOX_SetupConfigSections(void) {
         0
     };
 
-#if defined(__SSE__) && !(defined(_M_AMD64) || defined(__e2k__)) && !defined(EMSCRIPTEN)
-    CheckSSESupport();
+#ifdef __SSE__
+    CheckX86ExtensionsSupport();
 #endif
     SDLNetInited = false;
 

--- a/src/gui/render.cpp
+++ b/src/gui/render.cpp
@@ -125,21 +125,37 @@ static void RENDER_EmptyLineHandler(const void * src) {
 }
 
 /*HACK*/
-#if defined(__AVX2__)
+#ifdef __SSE__
+#ifdef __AVX2__
 /* We’re building with -mavx2 */
 # define sse2_available (1)
 # define avx2_available (1)
-#elif defined(__SSE__) && (defined(_M_AMD64) || defined(__amd64__) || defined(__e2k__))
+#elif defined(_M_AMD64) || defined(__amd64__) || defined(__e2k__)
 /* SSE2 is always available on x86_64 and Elbrus */
 # define sse2_available (1)
 extern bool             avx2_available;
 #else
-# ifdef __SSE__
 extern bool             sse2_available;
 extern bool             avx2_available;
-# endif
 #endif
 /*END HACK*/
+
+#ifdef __GNUC__
+__attribute__((__target__("avx2")))
+#endif
+static inline bool cacheHit_AVX2(const Bitu *src, Bitu *cache, Bits count) {
+    static const Bitu simd_inc = 32 / sizeof(*src);
+    while (count >= (Bits)simd_inc) {
+        __m256i v = _mm256_loadu_si256((const __m256i*)src);
+        __m256i c = _mm256_loadu_si256((const __m256i*)cache);
+        __m256i cmp = _mm256_cmpeq_epi32(v, c);
+        if (GCC_UNLIKELY((unsigned int)_mm256_movemask_epi8(cmp) != 0xFFFFFFFF))
+            return false;
+        count-=(Bits)simd_inc; src+=simd_inc; cache+=simd_inc;
+    }
+    return true;
+}
+#endif // __SSE__
 
 /* NTS: In normal conditions, the renderer at the start of the frame
  *      does not call the scaler but instead compares line by line
@@ -166,16 +182,9 @@ static inline bool RENDER_DrawLine_scanline_cacheHit(const void *s) {
         Bits count = (Bits)render.src.start;
 #if defined(__SSE__)
 #define MY_SIZEOF_INT_P sizeof(*src)
-        if (avx2_available) {
-            static const Bitu simd_inc = 32/MY_SIZEOF_INT_P;
-            while (count >= (Bits)simd_inc) {
-                __m256i v = _mm256_loadu_si256((const __m256i*)src);
-                __m256i c = _mm256_loadu_si256((const __m256i*)cache);
-                __m256i cmp = _mm256_cmpeq_epi32(v, c);
-                if (GCC_UNLIKELY((unsigned int)_mm256_movemask_epi8(cmp) != 0xFFFFFFFF))
-                    goto cacheMiss;
-                count-=(Bits)simd_inc; src+=simd_inc; cache+=simd_inc;
-            }
+        if (GCC_LIKELY(avx2_available)) {
+            if (!cacheHit_AVX2(src, cache, count))
+                goto cacheMiss;
         } else if (sse2_available) {
             static const Bitu simd_inc = 16/MY_SIZEOF_INT_P;
             while (count >= (Bits)simd_inc) {
@@ -189,7 +198,7 @@ static inline bool RENDER_DrawLine_scanline_cacheHit(const void *s) {
         }
 #undef MY_SIZEOF_INT_P
         else
-#endif
+#endif // __SSE__
         {
             while (count) {
                 if (GCC_UNLIKELY(src[0] != cache[0]))

